src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs

```diff
@@ -158,10 +158,6 @@ protected virtual void OnAfterArrange()
 
 			var result = _layouter.Measure(SizeFromUISize(availableSize));
 
-			result = IFrameworkElementHelper
-				.SizeThatFits(this, result)
-				.ToFoundationSize();
-
```

Comment:
Calling this resulted in a recursive call. 
`IFrameworkElementHelper.SizeThatFits(this, result)` will be calling internally `fe.XamlMeasure(availableSize);` see [here](https://github.com/unoplatform/uno/blob/6aa544c0e00a227cf68121786b9e56c0f46e47aa/src/Uno.UI/UI/Xaml/IFrameworkElement.cs#L303)

Reply:
Nice work tracking this down! You're right, this call should not be there, as it was remove from the similar location by @carldebilly in a recent clipping refactoring:

https://github.com/unoplatform/uno/blob/9eadb8ea4a8832c0abbb24656a9a809d26a36dfe/src/Uno.UI/UI/Xaml/FrameworkElement.iOS.cs#L111-L131

Both methods should be very close in structure, if not identical or even better shared.

Reply:
@jeromelaban 
Merged the iOS and macOS implementation leaving separate those methods that are different on each platform.